src/runtime/vdom/vdom-render.ts

```diff
@@ -629,36 +629,46 @@ export const patch = (oldVNode: d.VNode, newVNode: d.VNode) => {
   }
 };
 
+/**
+ * Adjust the `.hidden` property as-needed on any nodes in a DOM subtree which
+ * are slot fallbacks nodes.
+ *
+ * A slot fallback node should be visible by default. Then, it should be
+ * conditionally hidden if:
+ *
+ * - it has a sibling with a `slot` property set to its slot name or if
+ * - it is a default fallback slot node, in which case we hide if it has any
+ *   content
+ *
+ * @param elm the element of interest
+ */
 const updateFallbackSlotVisibility = (elm: d.RenderNode) => {
-  // tslint:disable-next-line: prefer-const
   const childNodes: d.RenderNode[] = elm.childNodes as any;
-  let childNode: d.RenderNode;
-  let i: number;
-  let ilen: number;
-  let j: number;
-  let slotNameAttr: string;
-  let nodeType: number;
-
-  for (i = 0, ilen = childNodes.length; i < ilen; i++) {
-    childNode = childNodes[i];
 
+  for (const childNode of childNodes) {
     if (childNode.nodeType === NODE_TYPE.ElementNode) {
       if (childNode['s-sr']) {
         // this is a slot fallback node
 
         // get the slot name for this slot reference node
-        slotNameAttr = childNode['s-sn'];
+        const slotName = childNode['s-sn'];
 
         // by default always show a fallback slot node
         // then hide it if there are other slots in the light dom
         childNode.hidden = false;
 
-        for (j = 0; j < ilen; j++) {
-          nodeType = childNodes[j].nodeType;
+        // we need to check all of its sibling nodes in order to see if
+        // `childNode` should be hidden
+        for (const siblingNode of childNodes) {
+          if (siblingNode === childNode) {
+            // same node, let's get outta here
+            continue;
+          }
```

Comment:
This looks like a change in actual functionality (please correct me if I'm wrong though!), can you help me understand why it's safe to do this/how we can verify that?

Reply:
haha good catch, I think I had the refactoring madness on this one and while I remember thinking I needed to add this I don't remember now why I thought so, but this check definitely wasn't there before.